### PR TITLE
#2085 Modified org.eclipse.smarthome.automation.commands to use DS

### PR DESCRIPTION
**Pull request by vilchev**
Signed-off-by: Vasil Ilchev <v.ilchev@prosyst.bg>

### CODE REVIEW

File: bundles/automation/org.eclipse.smarthome.automation.commands/OSGI-INF/AutomationCommandsPluggable.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~  Copyright (c) 2016 by ProSyst Software GmbH.,
+  ~  url: http://www.prosyst.com
+  ~  All rights reserved.
+  ~
+  ~  This software is the confidential and proprietary information
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.commands/OSGI-INF/AutomationCommandsPluggable.xml]: This header does not seem to be the correct one, right?

File: bundles/automation/org.eclipse.smarthome.automation.commands/OSGI-INF/AutomationCommandsPluggable.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" activate="activate" deactivate="deactivate"
+               immediate="true" name="org.eclipse.smarthome.automation.commands">
+
+    <implementation class="org.eclipse.smarthome.automation.internal.commands.AutomationCommandsPluggable"/>
+
+    <service>
+        <provide interface="org.eclipse.smarthome.io.console.extensions.ConsoleCommandExtension"/>
+    </service>
+
+    <reference bind="setRuleRegistry" cardinality="0..1"
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.commands/OSGI-INF/AutomationCommandsPluggable.xml]: Does this work when the services comes later? From my experience, if the component is satisfied without this service already, setRuleRegistry is not called afterwards anymore.
I guess we could change all three dependencies to 1..1 cardinality as well.

Comment (maggu2810) [inline bundles/automation/org.eclipse.smarthome.automation.commands/OSGI-INF/AutomationCommandsPluggable.xml]: It depends on the reference policy (static, dynamic) and the reference policy option (reluctant, greedy):
It should be ignored using "static reluctant" but works for the other (three) ones (static greedy, dynamic reluctant and dynamic greedy).

**Comment by vilchev** [comment]
I've Just updated the PR with fixed license header in the AutomationCommandsPluggable.xml

File: bundles/automation/org.eclipse.smarthome.automation.commands/OSGI-INF/AutomationCommandsPluggable.xml (hunk 2 above)
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.commands/OSGI-INF/AutomationCommandsPluggable.xml]: Afaik, greedy/reluctant doesn't exist in OSGi 4.2 - and if reluctant is the behavior in this case, I think we have to change it to 1..1.

Comment (maggu2810) [inline bundles/automation/org.eclipse.smarthome.automation.commands/OSGI-INF/AutomationCommandsPluggable.xml]: It seems you know it :wink: I found the option in 4.3 but at the moment I don't see it in the 4.2.

Regardless if it works changing the policy etc. using a cardinality of 1..1 makes sense, too.
So, let's go for this change.

Comment (vilchev) [inline bundles/automation/org.eclipse.smarthome.automation.commands/OSGI-INF/AutomationCommandsPluggable.xml]: Ok, I'll update the PR

**Comment by vilchev** [comment]
Travis build failed
[INFO] Eclipse SmartHome Automation Module Timer Tests ... FAILURE [15.495s]

This failure test does not have anything to do with the console commands PR (at least I cannot find logical link to the failure test).

What should I do?

**Comment by marinmitev** [comment]
The latest code looks good to me, please merge it.

**Comment by kaikreuzer** [comment]
Thx.

**Pull request merged**